pacman/operations/partition_algorithms/basic_partitioner.py

```diff
@@ -116,7 +119,7 @@ def _partition_one_application_vertex(
                 "{}:{}:{}".format(vertex.label, first, last),
                 get_remaining_constraints(vertex))
             m_graph.add_vertex(m_vertex)
-            mapper.add_vertex_mapping(m_vertex, vertex_slice, vertex)
+            vertex.remember_associated_machine_vertex(m_vertex)
```

Comment:
I wonder if it would not be better to have Machine_vertex init do the registrastion.

Otherwise it is sometimes in the partitioner and sometimes in create_machine_vertex (for example LivePacketGather)

Reply:
Or rather that you can do it when a machine vertex is added to its graph provided the machine vertex knows about its originating application vertex. Which it should do; it was made by it.